### ISSUE TITLE
Runplugin cli tool to support declaring a deployment name

### ISSUE THREAD

**Issue by gemblerz** [sagecontinuum/ses#7]
The cli tool uses plugin image name as a name for a Kubernetes deployment object. This limits the ability to run multiple deployments using the same plugin image, e.g., running image samplers using `waggle/plugin-image-sampler:0.1.1` for 2 cameras. The tool may need to accept the deployment name as an argument, like

```/bin/bash
# if -name is not present, it uses the image name for the deployment name
$ runplugin -name image-sampler-camera1 waggle/plugin-image-sampler:0.1.1 -stream camera1
$ runplugin -name image-sampler-camera2 waggle/plugin-image-sampler:0.1.1 -stream camera2
```

**Comment by seanshahkarami** [comment]
Instead of requiring the user / scheduler to come up with the explicit names, would you prefer add part of the sha sum of the yaml representation? That way any any change of params becomes a new unique name?

Something like deployments names like `plugin-image-sampler-48f7fd8`.

That way, if the same config is provided, you know you're already running it?

**Comment by seanshahkarami** [comment]
I should have been clear, the `48f7fd8` is something like the `gen yaml repr | take sha1 hash | take first 7 characters`.

**Comment by seanshahkarami** [comment]
I should have also said, I agree that this is a problem. For reference, I _think_ the suggestion I'm making is similar to how Kustomize handles a similar problem when adding mods to a yaml file.

**Comment by wgerlach** [comment]
Including the version might be useful , e.g. `plugin-image-sampler-0.1.1-48f7fd8` ?

**Comment by gemblerz** [comment]
Having the part of sha1 sounds reasonable to me. I also like Wolfgang's idea of including version number. The naming `PLUGIN-VERSION-INSTANCE` was the idea we had for differentiating plugins.

**Comment by gemblerz** [comment]
This has been resolved by #9
